src/common/mmap.c

```diff
@@ -521,31 +537,174 @@ util_range_is_pmem(const void *addr, size_t len)
 	do {
 		struct map_tracker *mt = util_range_find(addr, len);
 		if (mt == NULL) {
-			LOG(4, "address not found %p", addr);
+			LOG(4, "address not found 0x%016" PRIxPTR, addr);
 			retval = 0;
 			break;
 		}
 
-		LOG(10, "range found - begin %p end %p flags %x",
+		LOG(10, "range found - begin 0x%016" PRIxPTR
+				" end 0x%016" PRIxPTR
+				" flags %x",
 				mt->base_addr, mt->end_addr, mt->flags);
 
 		if (mt->base_addr > addr) {
-			LOG(10, "base address doesn't match: %p > %p",
+			LOG(10, "base address doesn't match: "
+				"0x%" PRIxPTR " > 0x%" PRIxPTR,
 					mt->base_addr, addr);
 			retval = 0;
 			break;
 		}
 
 		retval &= ((mt->flags & MTF_DIRECT_MAPPED) != 0);
 
-		uintptr_t map_len = ((uintptr_t)mt->end_addr - (uintptr_t)addr);
+		uintptr_t map_len = mt->end_addr - addr;
 		if (map_len > len)
 			map_len = len;
 		len -= map_len;
-		addr = (char *)addr + map_len;
+		addr += map_len;
 	} while (len > 0);
 
 	util_rwlock_unlock(&Mmap_list_lock);
 
 	return retval;
 }
+
+static int
+acquire_region_fd(int slash_dev, int regs_dir_fd, struct ndctl_ctx *ctx,
```

Comment:
this is linux specific, make a generic API and implement it in a different file.